File: front/bem_te_vi/lib/models/post.dart
class Post {
  final String id;
  final String description;
  final String authorId;
  final List<dynamic> likes;
  final List<dynamic> comments;

  Post({
    required this.id,
    required this.description,
    required this.authorId,
    required this.likes,
    required this.comments,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      description: json['description'],
      authorId: json['author'],
      likes: json['likes'],
      comments: json['comments'],
    );
  }
}

File: api/src/main/java/com/bemtevi/bem_te_vi_api/model/Post.java
package com.bemtevi.bem_te_vi_api.model;


import lombok.Data;
import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DBRef;
import java.time.LocalDateTime;
import java.util.List;
@Data
@Document(collection = "posts")
public class Post {
    @Id
    private String id;
    private String description;
    @Getter
    private byte[] image;
    private LocalDateTime createdAt;

    @DBRef
    private User author;

    private List<String> likes = List.of();

    @DBRef
    private List<Comment> comments = List.of();

    public int getLikeCounter() {
        return likes.size();
    }

    public int getCommentsCounter() {
        return comments.size();
    }
}
